_layouts/default.html

```diff
@@ -26,5 +26,11 @@
         </script>
       </amp-analytics>
     {% endif %}
+    {% if site.enable_pwa %}
+    <amp-install-serviceworker src="/sw.js"
+      data-iframe-src="{{ site.url }}/sw.html"
+      layout="nodisplay">
+    </amp-install-serviceworker>
+    {% endif %}
```

Comment:
`amp-install-serviceworker` is an AMP tag to do what it says; it's essentially performing an `iframe` hack on the same domain, which is normally not allowed per AMP.